bundles/binding/org.openhab.binding.ipx800/src/main/java/org/openhab/binding/ipx800/internal/Ipx800Binding.java

```diff
@@ -0,0 +1,299 @@
+				
+				
+				
+				Ipx800Item item;
+				if (port.getCommandType() == Ipx800PortType.OUPUT) {
+					boolean pulse = false;
+					if (config.getExtra(0).equals("p")) {
+						pulse = true;
+					}
+					item = new Ipx800OutputItem(pulse);
+				} else if (port.getCommandType() == Ipx800PortType.COUNTER) {
+					if (config.getExtra(0).equals("a")) {
+						float unit = 1;
+						if (!config.getExtra(1).equals("")) {
+							unit =  Float.parseFloat(config.getExtra(1));
+						}
+						Ipx800ConsumptionPeriod period = Ipx800ConsumptionPeriod.getPeriod(config.getExtra(2));
+						item = new Ipx800Consumption(unit, period);
+					} else {
+						item = new Ipx800Counter();
+					}
+				} else {
+					if (config.getExtra(0).equals("D")) {
+						item = new Ipx800DoubleClic();
+						port.switchToMultiHandler();
+					} else if (config.getExtra(0).equals("d")) {
+						item = new Ipx800SimpleClic();
+						port.switchToMultiHandler();
+					} else if (config.getExtra(0).equals("v")) {
+						if (config.getExtra(1).equals("")) {
+							item = new Ipx800Dimmer();
+						} else {
+							item = new Ipx800Dimmer(new Integer(config.getExtra(1)));
+						}
+						port.switchToMultiHandler();
+					} else if (config.getExtra(0).equals("m")) {
+						item = new Ipx800Mirror();
+					} else {
+						item = new Ipx800AstableSwitch();
+					}
+				}
+				item.setBinding(this);
+				item.setItemName(itemName);
+				
+				// If no to action set : null is returned
+				String toDeviceName = config.getToDeviceName();
+				if(toDeviceName.isEmpty()) {
+					toDeviceName = config.getDeviceName();
+				}
+				Tuple<Ipx800DeviceConnector, Ipx800Port> devToPort = getDeviceAndPort(toDeviceName, config.getToPortField());
+				Ipx800DeviceConnector toDevice = devToPort.x;
+				Ipx800Port toPort = devToPort.y;
+				Ipx800OutputItem toItem = null;
+				if (toDevice != null && toPort != null) {
+					toItem = new Ipx800OutputItem(config.isToPulse());
+					toItem.setItemName(itemName + "-out");
+					toItem.setBinding(this);
+					toItem.setFromItem(item);
+					item.setToItem(toItem);
+					toPort.attachItem(itemName + "-out", toItem);
+					
+				}
+				
+				port.attachItem(itemName, item);
+				if(toItem != null) {
+					logger.info("Item {} created using {}, attached to {} on port {} to item {}", itemName, item, device, port, toItem);
+				} else {
+					logger.info("Item {} created using {}, attached to {} on port {}", itemName, item, device, port);
+				}
+			}
+		}
+	}
+
+	public void activate() {
+		logger.debug("Activate called");
+	}
+
+	public void deactivate() {
+		for (Ipx800DeviceConnector device : devices.values()) {
+			device.destroyAndExit();
+		}
+		logger.debug("DeActivate called");
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		logger.debug("Received command item='{}', command='{}'", itemName, command.toString());
+		for (Ipx800DeviceConnector device : devices.values()) {
```

Comment:
the idea is to iterate over all BindingProviders ask them for proper configuration for a given itemName (see NtpBindingProvider and NtpBinding as example)